GSA-TTS/rails-template
# Easy sandbox deploys

# 🎫 Addresses issue: #130 

closes #130 

## 🛠 Summary of changes


* Create a `terraform.sh` script that can deploy a sandbox 
* Create a sandbox_bot module to automatically create a service account to run terraform.sh
* Updates to the terraform generator so that it can still work when being used in the `sandbox-gsa` organization (previously, everything would fail because you couldn't use `OrgManager` permissions) (this ended up being a bigger lift than I meant it to)

## 📜 Testing Plan

How would a peer test this work?

Similar to #128, create a new app and deploy with`./terraform.sh`

* checkout branch
* run: export RT_DEV=true
* run: exe/rails_template_18f new ../my-rails-app and answer the questions. use gsa-tts-devtools-prototyping as the cg org, and make up a new space name for prod and staging spaces. They should not yet exist.
* move over to the new app and follow the terraform readme to deploy the app to cloud.gov
* repeat as above, but use your `sandbox-gsa` org and space

Possible gotcha - in order to use the script to deploy to staging or production environments, you need to tell it to use a different secrets file than it expects. Do this with:

```bash
./terraform.sh -e staging -c apply -- -var-file=secrets.cicd.tfvars
```
